Description: Patch is applied to fix several build issues.
 Rename the resample executable to zita-resampler, an executable
 called 'resample' is provided by another package.
Author: Alessio Treglia <alessio@debian.org>
Author: Dennis Braun <d_braun@kabelmail.de>
Last-Update: 2021-01-01
Forwarded: not-needed

Index: zita-resampler/apps/Makefile
===================================================================
--- zita-resampler.orig/apps/Makefile
+++ zita-resampler/apps/Makefile
@@ -22,17 +22,16 @@ PREFIX ?= /usr/local
 BINDIR ?= $(PREFIX)/bin
 MANDIR ?= /usr/share/man/man1
 
-VERSION = 1.6.0
-CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\"
+VERSION = 1.8.0
+CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\" -I../debian/tmp/usr/include -I../source
 CXXFLAGS += -O2  -ffast-math -Wall
-CXXFLAGS += -march=native
 
 
-all:	zresample zretune zresample.1.gz zretune.1.gz
+all:	zresample zretune
 
 
 ZRESAMPLE_O = zresample.o audiofile.o dither.o
-zresample:	LDLIBS += -lzita-resampler -lsndfile -lrt
+zresample:	LDLIBS += ../source/libzita-resampler.so.1.8.0 -lsndfile -lrt
 zresample:	$(ZRESAMPLE_O)
 	$(CXX) $(LDFLAGS) -o $@ $(ZRESAMPLE_O) $(LDLIBS)
 $(ZRESAMPLE_O):
@@ -40,34 +39,25 @@ $(ZRESAMPLE_O):
 
 
 ZRETUNE_O = zretune.o audiofile.o dither.o
-zretune:	LDLIBS += -lzita-resampler -lsndfile -lrt
+zretune:	LDLIBS += ../source/libzita-resampler.so.1.8.0 -lsndfile -lrt
 zretune:	$(ZRETUNE_O)
 	$(CXX) $(LDFLAGS) -o $@ $(ZRETUNE_O) $(LDLIBS)
 $(ZRETUNE_O):
 -include $(ZRETUNE_O:%.o=%.d)
 
 
-zresample.1.gz:	zresample.1
-	gzip -c zresample.1 > zresample.1.gz
-
-zretune.1.gz:	zretune.1
-	gzip -c zretune.1 > zretune.1.gz
-
-
 install:	all
-	install -d $(BINDIR)
-	install -d $(MANDIR)
-	install -m 755 zresample $(DESTDIR)$(BINDIR)
-	install -m 755 zretune $(DESTDIR)$(BINDIR)
-	install -m 644 zresample.1.gz $(DESTDIR)$(MANDIR)
-	install -m 644 zretune.1.gz $(DESTDIR)$(MANDIR)
+	install -Dm 755 zresample $(DESTDIR)$(PREFIX)/bin/zita-resampler
+	install -Dm 755 zretune $(DESTDIR)$(PREFIX)/bin/zita-retune
+	install -Dm 644 zresample.1 $(DESTDIR)$(MANDIR)/zita-resampler.1
+	install -Dm 644 zretune.1 $(DESTDIR)$(MANDIR)/zita-retune.1
 
 uninstall: 
 	/bin/rm -f $(DESTDIR)$(BINDIR)/zresample
 	/bin/rm -f $(DESTDIR)$(BINDIR)/zretune
-	/bin/rm -f $(MANDIR)/zresample.1.gz
-	/bin/rm -f $(MANDIR)/zretune.1.gz
+	/bin/rm -f $(MANDIR)/zresample.1
+	/bin/rm -f $(MANDIR)/zretune.1
 
 clean:
-	/bin/rm -f *~ *.o *.a *.d *.so *.gz zresample zretune
+	/bin/rm -f *~ *.o *.a *.d *.so zresample zretune
 
Index: zita-resampler/source/Makefile
===================================================================
--- zita-resampler.orig/source/Makefile
+++ zita-resampler/source/Makefile
@@ -31,9 +31,7 @@ VERSION = $(MAJVERS).$(MINVERS)
 DISTDIR = zita-resampler-$(VERSION)
 
 CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
-CPPFLAGS += -DENABLE_SSE2
 CXXFLAGS += -Wall -fPIC -O2 -ffast-math
-CXXFLAGS += -march=native
 LDFLAGS += 
 LDLIBS +=
 
@@ -48,7 +46,7 @@ ZITA-RESAMPLER_H = zita-resampler/resamp
 
 
 $(ZITA-RESAMPLER_MIN): $(ZITA-RESAMPLER_O)
-	$(CXX) -shared $(LDFLAGS) -Wl,-soname,$(ZITA-RESAMPLER_MAJ) -o $(ZITA-RESAMPLER_MIN) $(ZITA-RESAMPLER_O) $(ZITA-RESAMPLER_DEP)
+	$(CXX) $(LDFLAGS) -shared -Wl,-soname,$(ZITA-RESAMPLER_MAJ) -o $(ZITA-RESAMPLER_MIN) $(ZITA-RESAMPLER_O) $(ZITA-RESAMPLER_DEP)
 
 $(ZITA-RESAMPLER_O):	$(ZITA-RESAMPLER_H)
 
@@ -58,8 +56,8 @@ install:	$(ZITA-RESAMPLER_MIN)
 	install -d $(DESTDIR)$(LIBDIR)
 	install -m 644 $(ZITA-RESAMPLER_H)   $(DESTDIR)$(INCDIR)/zita-resampler
 	install -m 755 $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)
-	ldconfig
-	ln -sf $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-RESAMPLER_SO)
+	ln -sf $(ZITA-RESAMPLER_MIN) $(DESTDIR)$(LIBDIR)/$(ZITA-RESAMPLER_MAJ)
+	ln -sf $(ZITA-RESAMPLER_MAJ) $(DESTDIR)$(LIBDIR)/$(ZITA-RESAMPLER_SO)
 
 uninstall:
 	/bin/rm -rf $(DESTDIR)$(INCDIR)/zita-resampler
